carma_utils/sensorlib/src/SemanticLidarSensor.py

```diff
@@ -0,0 +1,334 @@
+        Compute horizontal and vertical angular FOV extent from sensor perspective for each detected object.
+
+        :param detected_objects: List of objects currently considered for detection.
+        :return: Dictionary mapping actor ID to tuple of (horizontal, vertical) angular extents.
+        """
+        return dict([(detected_object.id,
+                      self.compute_actor_angular_extent(detected_object)) for detected_object in
+                     detected_objects])
+
+    def compute_actor_angular_extent(self, detected_object):
+        """
+        Compute the horizontal and vertical angular FOV extent from sensor perspective for the given object.
+
+        :param detected_object: Object under consideration.
+        :return: Tuple containing the maximal horizontal and
+        vertical angular extents consumed by the object, from the sensor's perspective.
+        """
+        corner_vectors_in_world_frame = detected_object.bounding_box_in_world_coordinate_frame
+        theta = [self.compute_horizontal_angular_offset(vec) for vec in corner_vectors_in_world_frame]
+        phi = [self.compute_vertical_angular_offset(vec) for vec in corner_vectors_in_world_frame]
+        horizontal_fov = max(theta) - min(theta)
+        vertical_fov = max(phi) - min(phi)
+        return horizontal_fov, vertical_fov
+
+    def compute_horizontal_angular_offset(self, vec):
+        """Compute horizontal angle of a vector in relation to the sensor, as measured from the x axis."""
+        p = vec - self.__sensor.position  # Position vector relative to sensor
+        return np.arctan2(p[1], p[0])
+
+    def compute_vertical_angular_offset(self, vec):
+        """Compute the vertical angle of a vector in relation to the sensor, as measured from the x-y plane."""
+        p = vec - self.__sensor.position
+        return np.arcsin(p[2] / np.linalg.norm(p))
+
+    def compute_adjusted_detection_thresholds(self, detected_objects, object_ranges):
+        """
+        Compute the adjusted detection threshold for each object, based on the object's range. This threshold varies
+        with distance to accommodate ray spreading which naturally leads to lower hitpoint counts, risking object
+        misdetections.
+        """
+        return dict([(detected_object.id,
+                      self.compute_adjusted_detection_threshold(object_ranges[detected_object.id]))
+                     for
+                     detected_object in detected_objects])
+
+    def compute_adjusted_detection_threshold(self, range):
+        """Compute the detection threshold appropriate for the given range."""
+        dt_dr = self.__simulated_sensor_config["detection_threshold_scaling_formula"][
+            "hitpoint_detection_ratio_threshold_per_meter_change_rate"]
+        t_nominal = self.__simulated_sensor_config["detection_threshold_scaling_formula"][
+            "nominal_hitpoint_detection_ratio_threshold"]
+
+        return dt_dr * range * t_nominal
```

Comment:
From the way the object's size is being represented by the horizontal and vertical fov degrees from the sensor's perspective, I no longer think we need the adjusted threshold. The threshold can stay just a fractional (0;1]